src/InfoMCStructHelper.cc

```diff
@@ -125,20 +131,25 @@ namespace mu2e {
     fillGenInfo(bestprimarysp, geninfo);
   }
 
+  
   void InfoMCStructHelper::fillGenInfo(const art::Ptr<SimParticle>& gp, GenInfo& geninfo) {
+    if(gp.isNonnull()){
+      fillGenInfo(*gp, geninfo);
+    }
+  }
+
+  void InfoMCStructHelper::fillGenInfo(const SimParticle& gp, GenInfo& geninfo) {
 
     GeomHandle<DetectorSystem> det;
 
-    if(gp.isNonnull()){
-      geninfo._pdg = gp->pdgId();
-      geninfo._gen = gp->creationCode();
-      geninfo._mom = gp->startMomentum().vect().mag();
-      geninfo._costh = std::cos(gp->startMomentum().vect().theta());
-      geninfo._phi = gp->startMomentum().vect().phi();
-      geninfo._pos = Geom::toXYZVec(det->toDetector(gp->startPosition()));
-      geninfo._time = gp->startGlobalTime(); 
+      geninfo._pdg = gp.pdgId();
```

Comment:
Should we consider renaming this struct, since it is no longer related to GenParticle?   I could also imagine a single fill function taking a specific level (0=direct, 1=parent, 2=grandparent, etc) as argument, using the same struct but different object and branch.  That would make the depth of genealogy stored in TrkAna configurable, which would be a nice feature.  That's probably beyond the intent of this PR.

Reply:
Yes, I think we whouls rename this struct to `SimInfo` to make it clear we are looking at information from `SimParticles`. I like the idea of a single function to fill this information so we can have a configurable genealogy depth for TrkAna.